migrations/00004-insert-fetchedDogs.js

```diff
@@ -0,0 +1,3005 @@
+exports.up = async (sql) => {
+  const fetchedDogs = [
+    {
+      bred_for: 'Small rodent hunting, lapdog',
+      breed_group: 'Toy',
+      height: {
+        imperial: '9 - 11.5',
+        metric: '23 - 29',
+      },
+      id: '1',
+      life_span: '10 - 12 years',
+      name: 'Affenpinscher',
+      origin: 'Germany, France',
+      temperament:
+        'Stubborn, Curious, Playful, Adventurous, Active, Fun-loving',
+      weight: {
+        imperial: '6 - 13',
+        metric: '3 - 6',
+      },
+      url: 'https://cdn2.thedogapi.com/images/Dkqdl0c6N.jpg',
+    },
+
+    {
+      bred_for: 'Coursing and hunting',
+      breed_group: 'Hound',
+      country_code: 'AG',
+      height: {
+        imperial: '25 - 27',
+        metric: '64 - 69',
+      },
+      id: '2',
+      life_span: '10 - 13 years',
+      name: 'Afghan-Hound',
+      origin: 'Afghanistan, Iran, Pakistan',
+      temperament: 'Aloof, Clownish, Dignified, Independent, Happy',
+      weight: {
+        imperial: '50 - 60',
+        metric: '23 - 27',
+      },
+      url: 'https://cdn2.thedogapi.com/images/4WiUpv_4W.jpg',
```

Comment:
If you're getting these images from a separate API, an alternative would be to write some further JavaScript in this file, so that you don't have to add all of the images by hand.

The JavaScript could do something similar to the following:

1. Loop over the `fetchedDogs` array
2. Fetch the image URLs for the current object in the array
3. Add the image to the current object

Reply:
If you decide to do this, you will probably need to control the maximum number of requests that are happening at the same time.

This is because if you just `.map` over an array normally, it will not wait for the promise to resolve and all 150 dogs will be requested at once (150 connections to the API at the same time!)

There are a number of libraries that can control the concurrency, such as `p-map`:

https://www.npmjs.com/package/p-map

Reply:
Oh and you may also want to consider having the dog images saved locally on your site instead of using external URLs to other websites.